test/core/util/passthru_endpoint.c

```diff
@@ -83,11 +83,15 @@ static void me_write(grpc_exec_ctx *exec_ctx, grpc_endpoint *ep,
   if (m->parent->shutdown) {
     ok = false;
   } else if (m->on_read != NULL) {
-    gpr_slice_buffer_addn(m->on_read_out, slices->slices, slices->count);
+    for (size_t i = 0; i < slices->count; i++) {
+      gpr_slice_buffer_add(&m->read_buffer, gpr_slice_ref(slices->slices[i]));
```

Comment:
the buffer has changed from `m->on_read_out` to `m->read_buffer`. Is this intentional?